src/devices/src/virtio/balloon/utils.rs

```diff
@@ -270,4 +270,58 @@ mod tests {
             RemoveRegionError::MmapFail(_)
         );
     }
+
+    /// -------------------------------------
+    /// BEGIN PROPERTY BASED TESTING 
+
+    use proptest::prelude::*;
+
+    fn random_pfn_16k_u32_max() -> impl Strategy<Value = Vec<u32>> {
```

Comment:
Why do these functions have an array size of 16K? The PFN buffer has a size of 2K, as the `MAX_PAGE_COMPACT_BUFFER` constant suggests. Shouldn't we test with this constant instead?